src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -109,4 +110,19 @@ public static void requireExactlyOneOf(Set<Modifier> modifiers, Modifier... mutu
   public static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
     return DEFAULT != null && modifiers.contains(DEFAULT);
   }
+
+  public static String identifier(String part) {
```

Comment:
I mean, if s/o produces syntax errors ... the compiler will tell anyway.
* `$T$T` will always crash the compiler
* `$T...` can't be used to declare the last method parameter in JavaPoet-style? And if it's found inside a generic `CodeBlock` ... well.